### PR TITLE
Add support for custom streams and paths

### PR DESCRIPTION
**Pull request by lisachenko**
This PR introduces support for different stream wrappers, such as http://, phar://, compress.zlib://, etc. This PR is experimental now because `realpath()` and `stream_resolve_include_path()` functions were replaced with custom resolver, need to add test cases before merging into master to verify, that everything is ok.

After merging of this PR it will be possible to use goaop for standalone console applications and tools that can be very interesting.

### CODE REVIEW

**Comment by lisachenko** [comment]
PR for #117

File: src/Go/Core/AspectKernel.php [hunk 1]
```diff
@@ -153,12 +155,32 @@ protected function getDefaultOptions()
...
             'containerClass'     => static::$containerClass,
         );
     }
 
+
+    /**
+     * Normalizes options for the kernel
+     * @param $options
+     *
+     * @return array|mixed
```
Comment (cordoval) [inline src/Go/Core/AspectKernel.php]: is this not always an array? why mixed?

File: src/Go/Core/GoAspectContainer.php [hunk 2]
```diff
@@ -186,6 +186,6 @@ public function isFresh($timestamp)
         if (!$this->maxTimestamp && $this->resources) {
             $this->maxTimestamp = max(array_map('filemtime', $this->resources));
         }
-        return $this->maxTimestamp < $timestamp;
```
Comment (cordoval) [inline src/Go/Core/GoAspectContainer.php:197]: leave a line break before the return

File: src/Go/Instrument/ClassLoading/SourceTransformingLoader.php [hunk 3]
```diff
@@ -97,23 +90,17 @@ public function filter($in, $out, &$consumed, $closing)
...
             $this->transformCode($metadata);
-            $this->bucket->data = $metadata->source;
-            $this->bucket->datalen = strlen($this->bucket->data);
-            if (!empty($this->bucket->data)) {
-                stream_bucket_append($out, $this->bucket);
-            }
+
+            $bucket = stream_bucket_new($this->stream, $metadata->source);
+            stream_bucket_append($out, $bucket);
             return PSFS_PASS_ON;
```
Comment (cordoval) [inline src/Go/Instrument/ClassLoading/SourceTransformingLoader.php:108]: would be more readable to leave a line before the return

File: src/Go/Instrument/PathResolver.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Go! OOP&AOP PHP framework
```
Comment (cordoval) [inline src/Go/Instrument/PathResolver.php]: I think you should change everything to AOP PHP Framework

File: src/Go/Core/AspectKernel.php (hunk 1 above)
Comment (lisachenko) [inline src/Go/Core/AspectKernel.php]: Friendly phpStorm add this for me :) This should be only "array"

Comment (cordoval) [inline src/Go/Core/AspectKernel.php]: maybe adjust it and plug your friendly theme as well http://github.com/cordoval/symfony2Colors

File: src/Go/Instrument/ClassLoading/SourceTransformingLoader.php (hunk 3 above)
Comment (lisachenko) [inline src/Go/Instrument/ClassLoading/SourceTransformingLoader.php:108]: sounds good, need to clean the source code and make a single line before returns.

File: src/Go/Instrument/PathResolver.php (hunk 4 above)
Comment (lisachenko) [inline src/Go/Instrument/PathResolver.php]: Created #119

File: src/Go/Core/AspectKernel.php (hunk 1 above)
Comment (lisachenko) [inline src/Go/Core/AspectKernel.php]: done :)

**Comment by lisachenko** [comment]
@cordoval you can try to play with https://github.com/lisachenko/gush/commit/64ad7d80eaf7b20f56097cb24233ae8c07b15530. It's initial configuration that simple prevent execution of gush itself :)

**Comment by matthiasnoback** [comment]
@lisachenko Interesting work!

**Comment by lisachenko** [comment]
@matthiasnoback yes, it was very interesting to solve this outstanding task :) There was a lot of magic and it took all my knowledge about streams to implement this properly. Hope, that I didn't make an error somewhere...

Thank you for your interest to my work :)

**Pull request merged**